src/librustc/traits/select.rs

```diff
@@ -2081,6 +2125,11 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
                 substs.upvar_tys(def_id, self.tcx()).collect()
             }
 
+            ty::TyGenerator(def_id, ref substs, interior) => {
+                let witness = iter::once(interior.witness());
```

Comment:
Shouldn't you be using the list of generator interior types? I mean, nobody should be having non-trivial OIBIT impls for tuples, but there's no reason to use them.